src/types.ts

```diff
@@ -75,7 +76,26 @@ export interface I18nConfig<C extends Context = Context> {
   fluentBundleOptions?: FluentBundleOptions;
   /**
    * An optional function that determines which locale to use.
-   * See [Locale Negotiation](https://github.com/grammyjs/i18n#locale-negotiation) for more details.
+   * See [Locale Negotiation](https://grammy.dev/plugins/i18n.html#custom-locale-negotiation) for more details.
    */
   localeNegotiator?: LocaleNegotiator<C>;
+  /**
+   * Convenience function for defining global variables that are used frequently in the translation context.
+   * Variables defined inside this can be used directly in the translation source file without having to specifying them when calling the translate function.
+   * It is possible to overwrite the values by re-defining them in the translation context of translate function.
+   *
+   * @example
+   * ```ts
+   * function defaultTranslationContext(ctx: Context) {
+   *   return {
+   *     name: ctx.from?.first_name!,
```

Comment:
`!` is a bad sign here. Are you sure that fluent will not throw an error if it is undefined?
```suggestion
   *     name: ctx.from?.first_name || 'No Name',
```
That's also not ideal, would be mush better to pass a localized value here

Reply:
> Are you sure that fluent will not throw an error if it is undefined?

Yes. Because `FluentVariable` won't support undefined.

> That's also not ideal, would be mush better to pass a localized value here

I don't think so. Developer should only be using `name` in their translations if its there. + They can always re-define it when calling `t`.

Reply:
> Yes. Because FluentVariable won't support undefined.

I think you might have misunderstood me. If it is not allowed to pass `undefined` to `FluentVariable` but we're passing it anyway and ignoring the error with `!` (which is roughly equivalent to `as string`), what would fluent do? Throw an error?

Reply:
Nope. No errors as I've tested at least.

Reply:
> I don't think so. Developer should only be using name in their translations if its there. + They can always re-define it when calling t.

That's a bad point. The whole idea of this feature is to provide a way to define global variables that you don't have to pass on per-call basis. It can be useful to be able to use `ctx.t` here. And it is already possible, right? I don't think we should add it to this example anyway.

Reply:
> Nope. No errors as I've tested at least.

Okay. But anyway I'd rather avoid such incorrect type casts and change it to something like `|| ''` instead

Reply:
> it is already possible, right?

I think so.

> That's a bad point.

You're right and I get your point. But just to ask, why would someone use `name` in a context that it does not exists 😄 

> I don't think we should add it to this example anyway.

Agreed. Any replacement to suggest? So that the example will show off a little bit more about the feature.

Reply:
> Okay. But anyway I'd rather avoid such incorrect type casts and change it to something like || '' instead

One point to mention, just `Context` has most of its properties (that we may use as values in translation) as _maybe_ undefined. Every property would need an optional value, which will likely never get used because the user know what exists in a context.

Reply:
> Any replacement to suggest?
`|| ''`

Reply:
Will update to that. Thanks

Reply:
We can even provide that variable only if ctx.from is defined. So user gets an error if they try to use it when it's undefines

Reply:
But I don't know how to write it elegantly

Reply:
The underlying problem is that we're mixing two unrelated concepts: context objects (event for message handling) and arguments for string interpolation.

What do you think about doing something like `ctx.values` which contains the same data as is currently accessible through `CTX` but works in a way we control? We would provide a function similar to `CTX` but it would fetch data from `ctx.values` (name subject to bike-shedding).

For example, if an absent value is accessed, we are able to throw a helpful error. Also, we'd have a clean separation of concerns.

Do you think this would be an improvement over the current design?